### PR TITLE
stdout/stderr flush on exit

### PR DESCRIPTION
**Pull request by rphillips**

### CODE REVIEW

File: deps/process.lua [hunk 1]
```diff
@@ -108,13 +108,31 @@ local function removeListener(self, _type, listener)
...
+
+  local function onCompleted()
+    if completed == 2 then
+      self:emit('exit', code)
+      os.exit(code)
+    end
+  end
+
+  local function onStdoutEnd()
+    completed = completed + 1
```
Comment (creationix) [inline deps/process.lua]: You should move the increment into the shared `onCompleted()` function since it's shared among all paths.

Comment (creationix) [inline deps/process.lua]: In fact, you can then just have a single shared `onEnd` callback that you directly use for both the streams.

Comment (creationix) [inline deps/process.lua]:
```lua
local left = 2
local function onEnd()
  left = left - 1
  if left > 0 then return end
  self:emit('exit', code)
  os.exit(code)
end

process.stdout:once('end', onEnd)
process.stderr:once('end', onEnd)
```

Comment (rphillips) [inline deps/process.lua]: good call. tweaked

**Comment by creationix** [comment]
+1

**Pull request merged**